### PR TITLE
Removed unused imports from files

### LINKED ISSUE #3483: Investigate Python Unused Imports
There are [a lot](https://lgtm.com/query/7596990423930761565/) of unused imports which clutter things.

### Describe the problem that you'd like solved

Refactoring to remove all unused imports.

### Proposal & Constraints

Remove all ununsed imports. This will reduce technical debt, decluttering things to make them clearer.

### Additional context

UPDATE:
As per @cclauss's comments below, for detecting unused imports, we should really be running
```
python3 -m pip install flake8
flake8 --select=F401
```

### Stakeholders

@cclauss @imskr

### PR DESCRIPTION
**Pull request by Yashs911**
Related to #3483 

refactor removed unused imports will commit 2 files at a time
### Stakeholders
@cclauss @imskr

### CODE REVIEW

**Comment by cclauss** [comment]
Modified commit message because this PR would not get rid of ALL unused imports so  #3483 should remain open even if this PR lands.

**Comment by Yashs911** [comment]
@cclauss  I have removed unused imports till .\openlibrary\plugins\openlibrary\utils.py so should I create a PR and commit them 2 at a time? or create separate PR for each of them?

**Comment by Yashs911** [comment]
@cclauss Travis CI seems unhappy with batch6/7/8/9/10

**Comment by cclauss** [comment]
https://travis-ci.com/github/internetarchive/openlibrary/jobs/396968206#L566

**Comment by Yashs911** [comment]
@cclauss can you help me with what all changes do I need to make. It says the line is too long but I actually removed a word so I am a little clueless here.
I am sorry for any inconvenience I may have caused you.

**Comment by cclauss** [comment]
Once you change a line, that line must comply with strict `flake8 --max-line-length=88` for the tests to pass.  This means that you need to wrap that line so that it is no longer than 88 characters max.  Changing max-line-length to 113 is not acceptable.

File: openlibrary/catalog/marc/parse.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 from openlibrary.catalog.marc.get_subjects import subjects_for_work
 from openlibrary.catalog.marc.marc_base import BadMARC, NoTitle, MarcException
-from openlibrary.catalog.utils import pick_first_date, tidy_isbn, flip_name, remove_trailing_dot, remove_trailing_number_dot
+from openlibrary.catalog.utils import pick_first_date, tidy_isbn, remove_trailing_dot, remove_trailing_number_dot
```
Comment (cclauss) [inline openlibrary/catalog/marc/parse.py:5]:
```suggestion
from openlibrary.catalog.utils import (
    pick_first_date, tidy_isbn, remove_trailing_dot, remove_trailing_number_dot
)
```

**Comment by Yashs911** [comment]
@cclauss I have done the required changes and added some new commits. Status check is missing from initial 6 commits so did I do something wrong?

**Comment by cclauss** [comment]
I still have concerns about https://github.com/internetarchive/openlibrary/issues/3483#issuecomment-705523903

**Comment by Yashs911** [comment]
@cclauss so should I create PR for each commit separately. But if the build is successful then there shouldn't be any problem??

**Comment by cclauss** [comment]
> If the build is successful then there shouldn't be any problem??

For that, the integration tests would need to be running https://github.com/internetarchive/openlibrary/pull/3682 and probably added to significantly.

Can you please create a subset of this PR that only removes Python standard library imports?  That should be safer and once that lands, we can rebase this PR on master and land the non-standard library changes.

**Comment by Yashs911** [comment]
@cclauss Is there a way I can automatically check for only python standard library unused imports, or do I have to do that manually?

**Comment by cclauss** [comment]
Manual I believe.

**Comment by Yashs911** [comment]
> Manual I believe. Do you use git on the command-line?

@cclauss Yes
I have created a new PR #3909 in which I have removed all python standard library unused imports till batch13

**Comment by mekarpeles** [comment]
@Yashs911 I'd strongly suggest breaking this PR into a few smaller PRs so we can help you merge them much more quickly :)

Some of these changes are very straight forward (like the unused standard libraries: `simplejson`, `re` etc.)

Some of the Open Library specific imports may have strange side effects which needs to be tested (I am not sure I blindly trust the linter in these cases)

**Comment by mekarpeles** [comment]
Thank you @cclauss for investing time in this one <3

**Comment by Yashs911** [comment]
@mekarpeles Do I still need to break this PR into smaller PR or after #3909 this can be rebased as suggested by @cclauss .

File: openlibrary/core/helpers.py [hunk 2]
```diff
@@ -32,7 +32,6 @@
 from infogami.utils.view import safeint
 
 # TODO: i18n should be moved to core or infogami
-from openlibrary.i18n import gettext as _
```
[the remark this answers is not in the document]
Comment (cclauss) [reply openlibrary/core/helpers.py:35]: Thanks.  `_()` is not called in this file and a line like 69 (`_, host, _, _, _ = urlsplit(href)`) is going to shadow its value.

I will create a PR to remove the comment.

**Pull request merged**